### ISSUE TITLE
The accuracy is lower than the results reported in the paper.

### ISSUE THREAD

**Issue by LemonQC** [sjtuytc/Neurips21-ProTo-Program-guided-Transformers-for-Program-guided-Tasks#7]
The arguments are fixed.
we run ``python run_experiment.py --do_train", which means that we use **trainval_unbiased_all_fully** and ``testdev_pred" ( The data is directly from Meta)。

The results are as follows. We only get 55% (vs 65 in paper). Is there any wrong or I miss some important details?
![image](https://user-images.githubusercontent.com/30914380/204170118-976e43dd-5696-4ece-a79e-619d98c1402e.png)

**Comment by sjtuytc** [comment]
You can reproduce some basic results via the mnmm data and the current code. But if you wish to reproduce our best results in the leaderboard，you need to check the following. 
First，our paper reports results on the test split. I don't think you can get these results locally.
Second，I don't think the numbers on the validation sets are so low. Did you change the batch size or train long enough epochs？
Third，if you check the mnmm's code，I remember that they treat the final answers as middle results. This is different from ours. You need to write additional codes to generate the middle results of routines.
Forth，some training data is simply wrong. If you look at the training data，you would find out that some occasionally points to nonexisting objects. We did a lot of dirty work to make the training data right. Especially the original data is wrong in some classes so their performances are poor.
Fifth，the original programs use different terms to represent the same meanings. For example，they use on，on the top of， to the top of，... to represent the same meaning. We made them into uniform forms to achieve good results.

**Comment by LemonQC** [comment]
> You can reproduce some basic results via the mnmm data and the current code. But if you wish to reproduce our best results in the leaderboard，you need to check the following. First，our paper reports results on the test split. I don't think you can get these results locally. Second，I don't think the numbers on the validation sets are so low. Did you change the batch size or train long enough epochs？ Third，if you check the mnmm's code，I remember that they treat the final answers as middle results. This is different from ours. You need to write additional codes to generate the middle results of routines. Forth，some training data is simply wrong. If you look at the training data，you would find out that some occasionally points to nonexisting objects. We did a lot of dirty work to make the training data right. Especially the original data is wrong in some classes so their performances are poor. Fifth，the original programs use different terms to represent the same meanings. For example，they use on，on the top of， to the top of，... to represent the same meaning. We made them into uniform forms to achieve good results.

OK, I will have a try. Is there any pre-trained model?

**Comment by sjtuytc** [comment]
I don't have pretrained models at this moment because I have changed my research interests. Sorry for that.

**Comment by LemonQC** [comment]
Ok,Thanks.

I can not find the results in the board

![image](https://user-images.githubusercontent.com/30914380/204177629-15e48767-ad84-41fd-a455-b995e1620ee0.png)

**Comment by Leavelk** [comment]
> 

Hello, have you fully reproduced the results of the paper now? What GPU are you using and how long is the training time?

**Comment by sjtuytc** [comment]
> Ok,Thanks.
> 
> I can not find the results in the board
> 
> ![image](https://user-images.githubusercontent.com/30914380/204177629-15e48767-ad84-41fd-a455-b995e1620ee0.png)

We did not make the results publically available because our results were much lower than the transformer-based SOTA at that time. But please be aware that the SOTA often uses cross-trained features on external datasets such as LXMERT.

**Comment by sjtuytc** [comment]
> > 
> 
> Hello, have you fully reproduced the results of the paper now? What GPU are you using and how long is the training time?

Eight 2080Ti should be fine within five days.

**Comment by LemonQC** [comment]
> lts publically available because our results were much lower than the transformer-based SOT

but how can you obtain the test results without submission？

**Comment by sjtuytc** [comment]
It's a good question. Almost all the competitions I know, including the GQA competition, can select both visible and non-visible competition. This happens after submission.

**Comment by LemonQC** [comment]
> It's a good question. Almost all the competitions I know, including the GQA competition, can select both visible and non-visible competition. This happens after submission.

OK，I see.
